advanced_source/cpp_export.rst

```diff
@@ -0,0 +1,364 @@
+distribution. If all goes well, it will look something like this:
+
+.. code-block:: sh
+
+  root@4b5a67132e81:/example-app# mkdir build
+  root@4b5a67132e81:/example-app# cd build
+  root@4b5a67132e81:/example-app/build# cmake -DCMAKE_PREFIX_PATH=/path/to/libtorch ..
+  -- The C compiler identification is GNU 5.4.0
+  -- The CXX compiler identification is GNU 5.4.0
+  -- Check for working C compiler: /usr/bin/cc
+  -- Check for working C compiler: /usr/bin/cc -- works
+  -- Detecting C compiler ABI info
+  -- Detecting C compiler ABI info - done
+  -- Detecting C compile features
+  -- Detecting C compile features - done
+  -- Check for working CXX compiler: /usr/bin/c++
+  -- Check for working CXX compiler: /usr/bin/c++ -- works
+  -- Detecting CXX compiler ABI info
+  -- Detecting CXX compiler ABI info - done
+  -- Detecting CXX compile features
+  -- Detecting CXX compile features - done
+  -- Looking for pthread.h
+  -- Looking for pthread.h - found
+  -- Looking for pthread_create
+  -- Looking for pthread_create - not found
+  -- Looking for pthread_create in pthreads
+  -- Looking for pthread_create in pthreads - not found
+  -- Looking for pthread_create in pthread
+  -- Looking for pthread_create in pthread - found
+  -- Found Threads: TRUE
+  -- Configuring done
+  -- Generating done
+  -- Build files have been written to: /example-app/build
+  root@4b5a67132e81:/example-app/build# make
+  Scanning dependencies of target example-app
+  [ 50%] Building CXX object CMakeFiles/example-app.dir/example-app.cpp.o
+  [100%] Linking CXX executable example-app
+  [100%] Built target example-app
+
+If we supply the path to the serialized ``ResNet18`` model we created earlier
+to the resulting ``example-app`` binary, we should be rewarded with a friendly
+"ok":
+
+.. code-block:: sh
+
+  root@4b5a67132e81:/example-app/build# ./example-app model.pt
+  ok
+
+Step 4: Executing the Script Module in C++
+----------------------------------------
+
+Having succesfully loaded our serialized ``ResNet18`` in C++, we are now just a
+couple lines of code away from executing it! Let's add those lines to our C++
+application's ``main()`` function:
+
+.. code-block:: cpp
+
+    // Create a vector of inputs.
+    std::vector<torch::jit::IValue> inputs;
+    inputs.push_back(torch::ones({1, 3, 224, 224}));
+
+    // Execute the model and turn its output into a tensor.
+    auto output = module->forward(inputs).toTensor();
+
+    std::cout << output.slice(/*dim=*/1, /*start=*/0, /*end=*/5) << '\n';
+
+The first two lines set up the inputs to our model. We create a vector of
+``torch::jit::IValue`` (a type-erased value type ``script::Module`` methods
+accept and return) and add a single input. To create the input tensor, we use
+``torch::ones()``, the equivalent to ``torch.ones`` in the C++ API.  We then
+run the ``script::Module``'s ``forward`` method, passing it the input vector we
+created. In return we get a new ``IValue``, which we convert to a tensor.
```

Comment:
"We need to call `toValue()` on the result because not all models will return a single tensor. Some might return a Tuple or a lists of tensors. IValue has methods to extract each from the result."